x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java

```diff
@@ -417,7 +417,7 @@ private class PeriodicMaintenanceTask implements Runnable, Releasable {
 
         private volatile Map<String, Set<String>> existingSnapshots;
         private volatile Set<String> existingRepositories;
-        private volatile SearchResponse searchResponse;
+        private final AtomicReference<SearchResponse> searchResponse = new AtomicReference<>();
```

Comment:
AFAICT this is all effectively single-threaded, so I'm not sure we need the extra protection that using `getAndSet()` gives us.

Also I'm pretty sure we don't need to retain the search response at all, we're really converting it directly into a bulk delete request as soon as we receive it. This `run()` method is doing too much.

Anyway nbd for now but would you open an issue to track this tech debt?

Reply:
https://github.com/elastic/elasticsearch/issues/103245 there we go